dask/tests/test_order.py

```diff
@@ -301,36 +299,24 @@ def test_run_smaller_sections(abcde):
      / \ /|  | /
     a   c e  cc
 
-    Prefer to run acb first because then we can get that out of the way
```

Comment:
This is currently favoring finishing `aa` before anything else. I believe this is a matter of finding the appropriate target functions.

Considering that it was xfailed I'm not terribly concerned but I will have a look again.

Reply:
This example is a bit ambiguous. There is a "best ordering" for single threaded execution and one for multi threaded execution.

Single threaded

This PR
| Step     | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
| -        | - | -- | -- | -- | - | - | - | - | - |
| T1       | e | c  | d | cc | bb | aa | dd | a | b |
| Pressure | 1 | 2  | 2  | 3  | 4 | 3  | 3  | 4 | 2 |

What the docs say

| Step     | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
| -        | - | -- | -- | -- | - | - | - | - | - |
| T1       | c | a  | b  | e  | d | cc | dd | bb | aa |
| Pressure | 1 | 2  | 2  | 3  | 2 | 3  | 3  | 3 | 2 |


Multi threaded

This PR

| Step     | 0 | 1 | 2 | 3 | 4 |
| -        | - | -- | -- | -- | - |
| T1       | e | d  | bb | aa | b |
| T2       | c | cc | dd | a  | x |
| Pressure | 2 | 3  | 3  | 2  | 2 |


What the docs say

| Step     | 0 | 1 | 2 | 3 | 4 |
| -        | - | -- | -- | -- | - |
| T1       | a | b | d  | bb | aa |
| T2       | c | e | cc | dd | x |
| Pressure | 2 | 2 | 3  | 4  | 2 |

Reply:
Typically dask.order acts as if we were executing sequentially so what the docs say is not wrong. However, in practice, most will be running in a multi threaded environment. This behavior may, of course, look again very different for more threads...